### PR TITLE
Options for precipitation_accumulation to compute solid and/or liquide phases

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated

### After merging to master and before closing the branch:
- [ ] bumpversion (minor / major / patch) has been called on `master` branch
- [ ] Tags have been pushed

* **What kind of change does this PR introduce?** 
Moves "precipitation_accumulation" to the `_multivariate.py` file. Adds two optional arguments to pass the mean temperature and phase selection. If `phase` is `solid`, only time steps where T < 0°C are summed. If it is `liquid`, time steps with T >= 0°C are summed. If it is not provided, goes back to the usual behaviour, summing everything.

* **Does this PR introduce a breaking change?** 
Nope

* **Other information**:
Tests have been added.
 `xclim.atmos.precip_accumulation` has been modified so the attributes reflect the change, however, I have not yet found a way to modify the given standard name. (There is an ambiguity if `phase` is "liquid" and it should be different for "solid".

### CODE REVIEW

**Comment by Zeitsperre** [comment]
_Passes Black and Flake8 compliance on first push to PR._

Yes. This is how it was all meant to be.

**Comment by aulemahal** [comment]
Ok, so here is one solution to the "standard_name" problem.
Now, in `xclim.atmos`, cf attributes can be given as callables. Just before calling `str.format`, the callable is called with the arguments are keyword arguments. 
Solves it with some generality, but makes quite ugly lines in the `atmos` definitions...

**Comment by aulemahal** [comment]
A test was added and the atmos-generated description was modified.

File: xclim/indices/_multivariate.py [hunk 1]
```diff
@@ -538,6 +539,72 @@ def liquid_precip_ratio(
...
+    When the mean temperature is over 0 degC, precipitatio is assumed to be liquid rain and snow otherwise.
+
+    Parameters
+    ----------
+    pr : xr.DataArray
+      Mean daily precipitation flux [Kg m-2 s-1] or [mm].
+    tas : xr.DataArray, optional
+      Mean daily temperature [℃] or [K]
+    phase : str, optional,
+      Which phase to consider, "liquid" or "solid", if None (default), both are considered.
```
Comment (huard) [inline xclim/indices/_multivariate.py]: default is set to "both", not None

File: xclim/indices/_multivariate.py [hunk 2]
```diff
@@ -538,6 +539,72 @@ def liquid_precip_ratio(
...
+    --------
+    The following would compute for each grid cell of file `pr_day.nc` the total
+    precipitation at the seasonal frequency, ie DJF, MAM, JJA, SON, DJF, etc.:
+
+    >>> pr_day = xr.open_dataset('pr_day.nc').pr
+    >>> prcp_tot_seasonal = precip_accumulation(pr_day, freq="QS-DEC")
+    """
+    freq = freq or "YS"
+
+    if phase != "both":
```
Comment (huard) [inline xclim/indices/_multivariate.py]: "both" or None ?

File: xclim/indices/_multivariate.py [hunk 3]
```diff
@@ -538,6 +539,72 @@ def liquid_precip_ratio(
...
+        tu = units.parse_units(tas.attrs["units"].replace("-", "**-"))
+        fu = "degC"
+        frz = 0
+        if fu != tu:
+            frz = units.convert(frz, fu, tu)
```
Comment (huard) [inline xclim/indices/_multivariate.py]: Can be simplified with `utils.convert_units_to`.

**Comment by huard** [comment]
I've proposed changes to the PR. Feel free to dismiss them.

File: xclim/atmos/_precip.py [hunk 4]
```diff
@@ -107,9 +107,20 @@ class PrTas(Indicator2D):
 precip_accumulation = Pr(
     identifier="prcptot",
     units="mm",
-    standard_name="lwe_thickness_of_precipitation_amount",
-    long_name="Total precipitation",
-    description="{freq} total precipitation.",
+    standard_name=lambda **kw: "lwe_thickness_of_snowfall_amount"
```
[the remark this answers is not in the document]
Comment (aulemahal) [reply xclim/atmos/_precip.py]: Makes sense!
So I could just write to new function that only call the main one.

File: xclim/indices/_multivariate.py (hunk 1 above)
Comment (aulemahal) [reply xclim/indices/_multivariate.py]: Woups. Scusi...

File: xclim/indices/_multivariate.py (hunk 2 above)
Comment (aulemahal) [reply xclim/indices/_multivariate.py]: This is a choice that we could make. What do you prefer?
For me "both" was more explicit.

File: xclim/atmos/_precip.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (huard) [reply xclim/atmos/_precip.py]: Yep, done in my commit.

File: xclim/indices/_multivariate.py (hunk 2 above)
Comment (huard) [reply xclim/indices/_multivariate.py]: Not a strong opinion on this. I think `None` is more universal.  @tlogan2000 ?

Comment (tlogan2000) [reply xclim/indices/_multivariate.py]: No strong opiinon either but I would maybe go with ```None``` to be consistent with some other default behavoir in xclim as a whole

Comment (aulemahal) [reply xclim/indices/_multivariate.py]: 2 vs 1, allons-y.

**Pull request merged**